validator/index.js

```diff
@@ -266,21 +460,41 @@ function main() {
 
   program.command('* <fileOrUrlOrMinus...>')
       .description('Validates list of files or urls (default).')
-      .action((validateOrUrlOrMinus) => {
-        if (validateOrUrlOrMinus.length == 0) {
+      .action((fileOrUrlOrMinus) => {
+        if (fileOrUrlOrMinus.length == 0) {
           program.outputHelp();
           process.exit(1);
         }
-        readFileOrDownload(
-            program.validator_js,
-            (validatorScript) => {
-              vm.runInThisContext(validatorScript);
-              validateFiles(validateOrUrlOrMinus);
-            },
-            (errorMessage) => {
-              console.error('Could not fetch validator.js: ' + errorMessage);
-              process.exitCode = 1;
-            });
+        for (const item of fileOrUrlOrMinus) {
+          let input;
+          if (item === '-') {
+            input = readFromStdin();
+          } else if (isHttpOrHttpsUrl(item)) {
+            input = readFromUrl(item);
+          } else {
+            input = readFromFile(item);
+          }
+          input
+              .then((data) => {
+                getInstance(program.validator_js)
+                    .then((validator) => {
+                      const validationResult = validator.validateString(data);
+                      logValidationResult(item, validationResult);
+                      if (validationResult.status !== 'PASS') {
+                        process.exitCode = 1;
+                      }
+                    })
+                    .catch((error) => {
+                      console.error(
```

Comment:
Seems I commented `stdin` on the wrong line. My meaning is that `readFromStdin`, `readFromUrl` and `readFromFile` should each handle cases where they cannot read the data and throw a descriptive error message. Then, in both this catch and the one under, they simply console out the message. Eg. the catch block below says "unable to fetch" when reading from `stdin`, or a file, not just for URL fetches.